src/lang/std/engineConnection.ts

```diff
@@ -358,13 +358,13 @@ class EngineConnection extends EventTarget {
           break
         case EngineConnectionStateType.Disconnecting:
         case EngineConnectionStateType.Disconnected:
-          // Reconnect if we have disconnected.
-          if (!this.isConnecting()) this.connect(true)
+          // Let other parts of the app handle the reconnect
           break
         default:
           if (this.isConnecting()) break
           // Means we never could do an initial connection. Reconnect everything.
-          if (!this.pingPongSpan.ping) this.connect(true)
+          if (!this.pingPongSpan.ping)
+            this.connect(this.engineCommandManager.disableWebRTC ? true : false)
```

Comment:
@lf94 This is what I did to fix the unit tests, because I wasn't able to replicate the original issue i.e. https://github.com/KittyCAD/modeling-app/issues/3265 would you mind double checking this doesn't cause other issues?

Reply:
Only seeing this comment now. Let's see what happens with the latest changes

Reply:
@lf94 replying to 
https://github.com/KittyCAD/modeling-app/pull/3250#issuecomment-2270336165

This is my fix for the failing artifact map unit tests

Reply:
Oh but it's broken again hmmm.